src/controllers/QueryController.ts

```diff
@@ -501,11 +503,11 @@ export class QueryController extends RootComponent {
   }
 
   public enableHistory() {
-    this.historyStore = new history.HistoryStore();
+    this.searchInterface.actionHistory = new history.HistoryStore();
   }
 
   public disableHistory() {
-    this.historyStore = new history.HistoryStore(new NullStorage());
+    this.searchInterface.actionHistory = new history.HistoryStore(new NullStorage());
```

Comment:
Do we need to clear the store somehow here before setting it to null storage ?

Reply:
If this method is intended to be called directly, I think that would be a good idea.

I implemented this method last year with the intent of calling it from the `disable` method in `Analytics.ts`. The reason that the storage isn't cleared here is because I cleared it in `Analytics.ts`'s `this.clearLocalData()` call inside the same `disable` method.

Reply:
Yep, as Benjamin said, there are separate methods on the `QueryController` to `clear` and `disable`. These are called from `Analytics`. We can adjust the behaviour of `disable` to make it more encapsulated, but I'm not sure if this could be breaking to someone?